Checks if schema attributes exist

hwinne opened the pull request:
## Because of some problems, it was easier to create a new PR. [The new PR can be found here](https://github.com/Yoast/wordpress-seo/pull/13789).

## Summary

Added `isset` checks for values which can be non existing, when they're left empty.

<!--
Attach one of the following labels to the PR: `changelog: bugfix`, `changelog: enhancement`, `changelog: other`, `changelog: non-user-facing`.
If the changelog item is a bugfix, please use the following sentence structure: Fixes a bug where ... would ... (when ...).
If the changelog item is meant for the changelog of another repo, start you changelog item with the repo name between square brackets, for example: * [wordpress-seo-premium] Fixes a bug where ....
If the same changelog item is applicable to multiple changelogs/repos, add a separate changelog item for all of them.
-->
This PR can be summarized in the following changelog entry:

* Fixes a bug where a notice would be thrown when fields in the FAQ or HowTo blocks were left empty.

## Relevant technical choices:

* Added checks to assure all required keys exist. If not, `return`.

## Test instructions
<!--
Please follow these guidelines when creating test instructions:
- Please provide step-by-step instructions how to reproduce the issue, if applicable.
- Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->
This PR can be tested by following these steps:

* Create a new post with an empty Faq block --> Publish.
    * No notices or warnings should be thrown.
    * Fill the Faq block with data, it should output this data and no warnings or notices should be thrown.

* Create a new post with an empty HowTo block --> Publish.
    * No notices or warnings should be thrown.
    * Fill the HowTo block with data, it should output this data and no warnings or notices should be thrown.


## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation
* [ ] I have written documentation for this change.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

johannadevos commented on src/generators/schema/faq.php:
```diff
@@ -72,12 +72,17 @@ public function is_needed( Meta_Tags_Context $context ) {
-			$graph[] = $this->generate_question_block( $question, $index, $context );
+		}
+		else {
+			return $graph;
```
The code could be simpler (less indented).

If you make the first if-statement a negation...
```
if ( ! isset( $context->blocks['yoast/faq-block']['attrs']['questions'] ) ) {
    return $graph;
}
```

... then, the rest of the original code can follow without being indented (and the `else` clause can disappear).

hwinne replied:
The FAQ-block does not seem to throw an error when it is empty. So this is not relative anymore.

hwinne commented:
It turns out the problem with the FAQ-block was not checking if an item existed. 

We were looking for the `questions` in `$context->blocks['yoast/faq-block']['attrs']['questions']`, while the blocks are put in a container (by us). So we should be looking in `$context->blocks['yoast/faq-block'][$blockNumber]['attrs']['questions']`. I changed the function, so we now loop through the `blocks['yoast/faq-block']` blocks. 

I also changed the logic in which the`numberOfItems` are counted, keeping in mind that there can only be one FAQ block.

johannadevos commented:
@hwinne This isn't a full code review, but I briefly looked through your code and think it would be better to rename `amount_of_blocks` to `number_of_blocks`, both for internal consistency and because blocks are countable, and therefore "number" is the better term :)

hwinne commented:
There were some problems, because it was easier to create a new PR, this PR will be closed. [The new PR can be found here](https://github.com/Yoast/wordpress-seo/pull/13789).
